File: OCP/Spoilers/MarsRoverKata/Commands/BackwardCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MarsRoverKata.Commands
{
    public class BackwardCommand : ICommand
    {
        public Rover Rover { get; set; }

        public BackwardCommand(Rover rover)
        {
            Rover = rover;
        }

        public bool Execute()
        {
            return Rover.MoveBackward();
        }
    }
}

File: LSP+ISP/Java_Exercise/MarsRoverKata/src/main/java/MarsRoverKata/Commands/BackwardCommand.java
package MarsRoverKata.Commands;

import MarsRoverKata.Rover;

public class BackwardCommand implements ICommand {
    private Rover rover;

    public Rover getRover() {
        return rover;
    }

    public void setRover(Rover rover) {
        this.rover = rover;
    }

    public BackwardCommand(Rover rover) {
        this.rover = rover;
    }

    public boolean execute() {
        return rover.moveBackward();
    }
}

